installer.sh

```diff
@@ -128,16 +128,24 @@ case "$ID" in
         esac
     ;;
 
-    fedora)
+    fedora | anolis)
         echo "${ID} selected."
         PACKAGE_MGR=$(command -v dnf)
         PYTHON_PREIN="python3 python3-devel python3-setuptools git wget patch"
-        PYTHON_DEPS="python3-pip gcc gcc-c++ openssl-devel swig python3-pyyaml python3-zmq python3-cryptography python3-tornado python3-requests python3-gpg yaml-cpp-devel procps-ng python3-psutil python3-lark-parser python3-alembic"
+        PYTHON_DEPS="python3-pip gcc gcc-c++ openssl-devel python3-pyyaml python3-zmq python3-cryptography python3-tornado python3-requests python3-gpg yaml-cpp-devel procps-ng python3-psutil python3-lark-parser python3-alembic"
         if [ "$(uname -m)" = "x86_64" ]; then
             PYTHON_DEPS+=" efivar-devel"
         fi
         BUILD_TOOLS="openssl-devel libtool make automake pkg-config m4 libgcrypt-devel autoconf autoconf-archive libcurl-devel libstdc++-devel uriparser-devel dbus-devel gnulib-devel doxygen libuuid-devel json-c-devel"
-        if [[ ${VERSION_ID} -ge 30 ]] ; then
+        if [ "${ID}" = "anolis" ]; then
+            pip3 install setuptools_rust
+            BUILD_TOOLS+=" rust cargo"
+        elif [ "${ID}" = "fedora" ]; then
+            PYTHON_DEPS+=" swig"
+        fi
+
+        MAJOR_VERSION_ID=$(echo ${VERSION_ID} | awk -F '.' '{print$1}')
+        if [[ ${MAJOR_VERSION_ID} -ge 30 ]] ; then
```

Comment:
this also is probably fedora specific

Reply:
At present, this judgment condition can also be met by anolis (because anolis currently doesn't have a version above 30), if this condition is not met in the future, a new judgment condition will be added.

Reply:
Better fix now than when it breaks and another PR is necessary.